scripts/lib.sh

```diff
@@ -20,13 +30,34 @@ query-mac() {
     [[ $(uname -s) = 'Darwin' ]]
 }
 
-query_big_sur() {
+query-big-sur() {
     if require sw_vers && sw_vers -productVersion | grep -E "11\." >/dev/null; then
         return 0
     fi
     return 1
 }
 
+query-apple-m1() {
+    query-mac && [[ $(uname -m) = 'arm64' ]]
+}
+
+query-pyenv-version() {
+    local PYENV_VERSION
+    PYENV_VERSION=3.6.13
+    query-apple-m1 && PYENV_VERSION=3.8.10 || :
+    echo "${PYENV_VERSION}"
+}
+
+# XXX: Fix this check to be more lineant
+query-valid-python-versions() {
+    python_version=$(python3 -V 2>&1 | awk '{print $2}')
+    if [ "${python_version}" == 3.6.13 ] || [ "${python_version}" == 3.8.10 ]; then
```

Comment:
I think it's better to be strict about it so we can keep people's devenvs as close as possible to production. Also, 3.8.11 is latest.

Reply:
Also, if we could hold off on merging this, that would be ideal since we're not yet ready yet for 3.8. Could separate some of your other refactors into separate PRs.